Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -420,13 +420,13 @@ extension TabTrayController: TabManagerDelegate {
 
     func tabManager(tabManager: TabManager, didRemoveTab tab: Browser, atIndex index: Int) {
         var newTab: Browser? = nil
-        self.collectionView.performBatchUpdates({ _ in
-            self.collectionView.deleteItemsAtIndexPaths([NSIndexPath(forItem: index, inSection: 0)])
-        }, completion: { finished in
-            if tabManager.count == 0 {
-                newTab = tabManager.addTab()
-            }
-        })
+
+        self.collectionView.deleteItemsAtIndexPaths([NSIndexPath(forItem: index, inSection: 0)])
+        self.collectionView.reloadItemsAtIndexPaths(self.collectionView.indexPathsForVisibleItems())
+
+        if tabManager.count == 0 {
+            newTab = tabManager.addTab()
```

Comment:
Should this be handled by the tabManager rather than the TabTrayController? I feel it is the managers job to ensure it has the right number of tabs rather than the controllers..